controllers/metal3.io/host_state_machine.go

```diff
@@ -161,7 +161,7 @@ func (hsm *hostStateMachine) checkDelayedHost(info *reconcileInfo) actionResult
 	// host not yet tracked by the provisioner
 	switch info.host.Status.Provisioning.State {
 	case metal3v1alpha1.StateInspecting, metal3v1alpha1.StateProvisioning,
-		metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
+		metal3v1alpha1.StateDeprovisioning:
```

Comment:
I guess then the `metal3v1alpha1.StateDeleting` should be removed also from the [recordActionDelayed](https://github.com/metal3-io/baremetal-operator/blob/c759a9d23f4eab33eb06ad5f5f74c0201551c71e/controllers/metal3.io/baremetalhost_controller.go#L339) for consistency

Reply:
It sounds also this [test case](https://github.com/metal3-io/baremetal-operator/blob/fa0829d7d7cd7d65a8f755f7a998167f85c8aeb8/controllers/metal3.io/host_state_machine_test.go#L190-L197) could be removed then